File: search-insert-position/search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int right = nums.size() - 1, left = 0, mid;
        while (left <= right) {
            mid = left + (right-left)/2;
            if (nums[mid] == target){
                return mid;
            }
            if (nums[mid] < target) {
                left = mid + 1;
            }
            else {
                right = mid - 1;
            }
        }
        return left;
    }
};

File: search-insert-position/search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if target in nums: return(nums.index(target))
        sorted(nums)
        myIndex = 0
        for n in nums:
            if n>target: 
                return(myIndex)
            myIndex+=1
        return(len(nums))